roles/clean_engine/tasks/remove_domain.yml

```diff
@@ -0,0 +1,16 @@
+# TODO: Check what happens when we force remove unattached storage domain (probably should add a default empty GUID as a data center
+# Answer: When we force remove an unattached storage domain, ansible tries to move it to maintenance and detach it first, although it might be that this storage domain is already detached and has no related data center, therefor the move to maintenance will fail
```

Comment:
when using 'absent', ovirt_storage_domains will call storage_domains_module.remove which will call _maintenance, and as part of the _maintenance method it will try to fetch the data center because it uses _attached_sds_service, and since the storage domain is not attached to anything it will fail
This is the failure that I get for using 'absent' on unattached storage domain:
Traceback (most recent call last):
  File "/tmp/ansible_jf3Kkc/ansible_module_ovirt_storage_domains.py", line 511, in main
    host=module.params['host'],
  File "/tmp/ansible_jf3Kkc/ansible_modlib.zip/ansible/module_utils/ovirt.py", line 648, in remove
    self.pre_remove(entity)
  File "/tmp/ansible_jf3Kkc/ansible_module_ovirt_storage_domains.py", line 391, in pre_remove
    self._maintenance(storage_domain)
  File "/tmp/ansible_jf3Kkc/ansible_module_ovirt_storage_domains.py", line 345, in _maintenance
    attached_sds_service = self._attached_sds_service()
  File "/tmp/ansible_jf3Kkc/ansible_module_ovirt_storage_domains.py", line 337, in _attached_sds_service
    dc = dcs_service.service(self._module.params['data_center']).get()
  File "/usr/lib64/python2.7/site-packages/ovirtsdk4/services.py", line 4031, in get
    return self._internal_get(headers, query, wait)
  File "/usr/lib64/python2.7/site-packages/ovirtsdk4/service.py", line 202, in _internal_get


I also uploaded a patch that fix that when using force:
https://github.com/ansible/ansible/pull/28962

Reply:
It broken from:

 https://github.com/ansible/ansible/pull/28659/files

When using `dc = dcs_service.service(self._module.params['data_center']).get()` we ge 404 instead of `None`,
we need to fix it in module.

Reply:
Thanks for clearing that out and sorry for the regression but something here look a bit weird, the method name is called _attached_sds_service that basically should mean that this storage domains must have a DC which it attached to.
so I think that this method should not be called at all for detached storage domains, no?

Reply:
No problem. Right so if `_attached_sds_service` returns None, it the DC isn't found so we can't return attached storage domain service, no?

Reply:
Proposed a fix in the following link: 
   https://github.com/ansible/ansible/pull/29175